utils/managed_bytes.hh

```diff
@@ -524,6 +524,10 @@ inline bool operator==(const managed_bytes_view& a, const managed_bytes_view& b)
     return a.size_bytes() == b.size_bytes() && compare_unsigned(a, b) == 0;
 }
 
+inline std::strong_ordering operator<=>(const managed_bytes_view& a, const managed_bytes_view& b) {
+    return compare_unsigned(a, b) <=> 0;
+}
```

Comment:
Well, this is risky, since managed_bytes is used to hold all sorts of data and this <=> is not correct for all data types. So it will be easy to misuse in situations where we need to defer to the type's dynamic compare operation.

Reply:
You could say the same for `bytes`, though, but that kind of comparison for `bytes` is in use. This change was needed to replace `nonwrapping_range<bytes>` with `nonwrapping_range<managed_bytes>` in `expression.cc`.

Reply:
That's unexpected, as expression.cc is supposed to use the data_type comparator, not byte operators.

Reply:
It's true that managed_bytes is both a container for bytes and an container for data that has to be interpreted by the type system.

 I don't see why nonwrapping_range requires operator<=>, it is explicitly for use with a supplied comparator. Perhaps it indicates a bug in nonwrapping_range or expression.cc.

Reply:
Hmm, I will check again where that operator<=> is used.

Reply:
I would recommend against having a comparison operator on ranges; it's almost guaranteed to confuse some users into thinking they can check for overlap with it.  But that's not true (see, eg, #8157).

Reply:
Okay, I did something weird. This operator really isn't used. I will remove it.
I would swear I wrote it because something wasn't compiling without it. I don't know what happened.

Reply:
Fixed.